gcloud-java-dns/src/main/java/com/google/gcloud/spi/DefaultDnsRpc.java

```diff
@@ -0,0 +1,178 @@
+    this.dns = new Dns.Builder(transport, new JacksonFactory(), initializer)
+        .setRootUrl(options.host())
+        .setApplicationName(options.applicationName())
+        .build();
+    this.options = options;
+  }
+
+  @Override
+  public ManagedZone create(ManagedZone zone) throws DnsException {
+    try {
+      return dns.managedZones().create(this.options.projectId(), zone).execute();
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public ManagedZone getZone(String zoneName, Map<Option, ?> options) throws DnsException {
+    // just fields option
+    try {
+      return dns.managedZones().get(this.options.projectId(), zoneName)
+          .setFields(FIELDS.getString(options)).execute();
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public Tuple<String, Iterable<ManagedZone>> listZones(Map<Option, ?> options)
+      throws DnsException {
+    // fields, page token, page size
+    try {
+      ManagedZonesListResponse zoneList = dns.managedZones().list(this.options.projectId())
+          .setFields(FIELDS.getString(options))
+          .setMaxResults(PAGE_SIZE.getInt(options))
+          .setPageToken(PAGE_TOKEN.getString(options))
+          .execute();
+      return Tuple.<String, Iterable<ManagedZone>>of(zoneList.getNextPageToken(),
+          zoneList.getManagedZones());
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public boolean deleteZone(String zoneName) throws DnsException {
+    try {
+      dns.managedZones().delete(this.options.projectId(), zoneName).execute();
+      return true;
+    } catch (IOException ex) {
+      DnsException serviceException = translate(ex);
+      if (serviceException.code() == HTTP_NOT_FOUND) {
+        return false;
+      }
+      throw serviceException;
+    }
+  }
+
+  @Override
+  public Tuple<String, Iterable<ResourceRecordSet>> listDnsRecords(String zoneName,
+      Map<Option, ?> options) throws DnsException {
+    // options are fields, page token, dns name, type
+    try {
+      ResourceRecordSetsListResponse response = dns.resourceRecordSets()
+          .list(this.options.projectId(), zoneName)
+          .setFields(FIELDS.getString(options))
+          .setPageToken(PAGE_TOKEN.getString(options))
+          .setMaxResults(PAGE_SIZE.getInt(options))
+          .setName(DNS_NAME.getString(options))
+          .setType(DNS_TYPE.getString(options))
+          .execute();
+      return Tuple.<String, Iterable<ResourceRecordSet>>of(response.getNextPageToken(),
+          response.getRrsets());
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public Project getProject(Map<Option, ?> options) throws DnsException {
+    try {
+      return dns.projects().get(this.options.projectId())
+          .setFields(FIELDS.getString(options)).execute();
+    } catch (IOException ex) {
+      throw translate(ex);
```

Comment:
Should we return `null` for not found or throw exception in this case?
I would return `null` in other get calls (for consistency with other modules) but here a NOT FOUND project seems to me a more noteworthy error.